### PR TITLE
[Fix](Nereids) fix leading hint should have all tables in one query block

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

example: when one query block have t1 join t2 join t3, leading should have these three tables together

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java [hunk 1]
```diff
@@ -287,12 +287,16 @@ public Long getTotalBitmap() {
...
-    public void setTotalBitmap() {
+    public void setTotalBitmap(Set<RelationId> inputRelationSets) {
         Long totalBitmap = 0L;
         if (hasSameName()) {
             this.setStatus(HintStatus.SYNTAX_ERROR);
             this.setErrorMessage("duplicated table");
         }
+        if (getTablelist().size() < inputRelationSets.size()) {
+            this.setStatus(HintStatus.SYNTAX_ERROR);
+            this.setErrorMessage("leading should have all tables in query block");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:298]: list missing table here?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java [hunk 2]
```diff
@@ -287,12 +287,16 @@ public Long getTotalBitmap() {
     /**
      * set total bitmap used in leading before we get into leading join
      */
-    public void setTotalBitmap() {
+    public void setTotalBitmap(Set<RelationId> inputRelationSets) {
         Long totalBitmap = 0L;
         if (hasSameName()) {
             this.setStatus(HintStatus.SYNTAX_ERROR);
             this.setErrorMessage("duplicated table");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:294]: list duplicated table name?

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java (hunk 2 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java]: done

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**